epoxy-annotations/src/main/java/com/airbnb/epoxy/EpoxyModelClass.java

```diff
@@ -1,16 +1,24 @@
 
 package com.airbnb.epoxy;
 
+import android.support.annotation.LayoutRes;
+
 import java.lang.annotation.ElementType;
 import java.lang.annotation.Retention;
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
 /**
- * Used to annotate EpoxyModel classes in order to generate a
- * subclass of that model with getters, setters, equals, and hashcode for the annotated fields.
+ * Used to annotate EpoxyModel classes in order to generate a subclass of that model with getters,
+ * setters, equals, and hashcode for the annotated fields, as well as other helper methods and
+ * boilerplate reduction.
  */
 @Target(ElementType.TYPE)
 @Retention(RetentionPolicy.CLASS)
 public @interface EpoxyModelClass {
+  /**
+   * A layout resource that should be used as the default layout for the model. If you set this you
+   * don't have to implement `getDefaultLayout`; it will be generated for you.
+   */
+  @LayoutRes int value() default 0;
```

Comment:
Hm, the easy option would be `@EpoxyModelClass(value=R.layout.hello_world, futureProperty=true)` which isn't great.

The other option is to rename the param from `value` to `layoutRes` in the future if we other params. That'd be a breaking change we'd have to force on people. Not too bad to upgrade if we provide a structural replace pattern, but also not great.

Another option is to create a whole new annotation. That could work depending on what the goal of the feature is.

Or we just do `@EpoxyModelClass(layoutRes=R.layout.hello_world)` now in the anticipation of adding future params.

That last option is the safest. It's not terrible to require the param name, but it would be a bit annoying. I supposed the param name would make the purpose clearer.

Thanks for bringing this up, hadn't considered it.